distribution/src/bin/elasticsearch

```diff
@@ -45,6 +50,19 @@ then
   fi
 fi
 
+if [[ $AUTO_CONFIG = true ]]; then
+  # ignore failures in auto-configuration
```

Comment:
I spoke with @colings86 about this today and we may want to revisit this decision.

Reply:
Which decision is that ?

Reply:
That if we fail to perform auto-configuration (can't write to file) that we might want to fail instead of ignore. Something about "leniency" and being "abhorent", blah blah blah :wink:

Reply:
I like aphorisms as the next person but this is not about leniency. We ( and by we I mean mostly @albertzaharovits ) have spent a lot of time thinking about the cases where we should fail and when we should not. This is written in code in #74868 and Albert has the task to write this down to the design doc too. In short, we consider the case where we can't write to the config file because it is set as read-only, as a case where we have an explicit configuration decision by the user ( the file doesn't get to be read only by its own). As such we have elected to not treat this as an error and fail to start the node ( and in doing so introduce a breaking change as we allow this today ) but simply not attempt to auto-configure security for them. Authentication will be enabled but we won't enable and configure TLS. Happy to reconsider or discuss alternative approaches, @colings86 feel free to bring this up for discussion in the project or we can chat anytime